catalog/format/iceberg/src/test/java/org/projectnessie/catalog/formats/iceberg/nessie/TestNessieModelIceberg.java

```diff
@@ -128,6 +142,114 @@ static Stream<Arguments> transformSpec() {
+
+    var fieldTimestamp =
+        Types.NestedField.of(4, false, "timestamp", Types.TimestampType.withoutZone());
+    var fieldTimestampDeep =
+        Types.NestedField.of(13, false, "deep_timestamp", Types.TimestampType.withZone());
+    var deeplyNestedStruct =
+        Types.NestedField.of(
+            10,
+            false,
+            "deeply",
+            Types.StructType.of(
+                Types.NestedField.of(11, true, "deep_dec", Types.DecimalType.of(10, 3)),
+                Types.NestedField.of(12, true, "deep_time", Types.TimeType.get()),
+                fieldTimestampDeep,
+                Types.NestedField.of(14, true, "deep_bin", Types.BinaryType.get()),
+                Types.NestedField.of(
+                    15,
+                    true,
+                    "deep_string",
+                    Types.ListType.ofRequired(16, Types.StringType.get()))));
+    var systemFields =
+        Types.StructType.of(
+            Types.NestedField.of(2, false, "partition", Types.IntegerType.get()),
+            Types.NestedField.of(3, false, "offset", Types.LongType.get()),
+            fieldTimestamp,
+            Types.NestedField.of(5, false, "headers", Types.ListType.ofRequired(6, headersKv)),
+            Types.NestedField.of(9, false, "key", Types.BinaryType.get()),
+            deeplyNestedStruct);
+
+    var fieldTestSchema = Types.NestedField.of(1, false, "test_schema", systemFields);
+    var schema = new Schema(List.of(fieldTestSchema));
+
+    var sourceName =
+        fieldTestSchema.name()
+            + '.'
+            + (firstLevel
+                ? fieldTimestamp.name()
+                : (deeplyNestedStruct.name() + '.' + fieldTimestampDeep.name()));
+
+    var spec =
+        PartitionSpec.builderFor(schema).withSpecId(42).day(sourceName, "daytime_day").build();
+    var sort = SortOrder.builderFor(schema).withOrderId(43).asc(sourceName).build();
+
+    var schemaJson = SchemaParser.toJson(schema);
+    var specJson = PartitionSpecParser.toJson(spec);
+    var sortJson = SortOrderParser.toJson(sort);
+
+    var nessieMapper = new ObjectMapper().findAndRegisterModules();
+    var icebergSchema = nessieMapper.readValue(schemaJson, IcebergSchema.class);
+    var icebergSpec = nessieMapper.readValue(specJson, IcebergPartitionSpec.class);
+    var icebergSort = nessieMapper.readValue(sortJson, IcebergSortOrder.class);
+
+    var icebergFields = new HashMap<Integer, NessieField>();
+    var nessieSchema = icebergSchemaToNessieSchema(icebergSchema, icebergFields);
+
+    var partitionFields1 = new HashMap<Integer, NessiePartitionField>();
+    soft.assertThatCode(
+            () -> icebergPartitionSpecToNessie(icebergSpec, partitionFields1, icebergFields))
+        .doesNotThrowAnyException();
+    soft.assertThatCode(() -> icebergSortOrderToNessie(icebergSort, icebergFields))
+        .doesNotThrowAnyException();
+
+    var collectedFields = collectFieldsByIcebergId(List.of(nessieSchema));
```

Comment:
I would expect `icebergFields` and `collectedFields` to have exactly the same size, but they are off by 2: fields 7 and 8 are missing in `collectedFields`. I think this is because they appear in a LIST type, and so they are not correctly collected.